src/azure-cli-core/azure/cli/core/keys.py

```diff
@@ -32,14 +35,34 @@ def is_valid_ssh_rsa_public_key(openssh_pubkey):
 def generate_ssh_keys(private_key_filepath, public_key_filepath):
     import paramiko
 
+    try:
+        with open(public_key_filepath, 'r') as public_key_file:
+            public_key = public_key_file.read()
+            pub_ssh_dir, _ = os.path.split(public_key_filepath)
+            logger.warning("Public SSH key file '%s' found in dir '%s'."
+                           "Use public key file. New RSA key pair will not be generated.",
+                           public_key_filepath, pub_ssh_dir)
+
+            return public_key
+    except IOError:
```

Comment:
`FileNotFoundError`, IOError spans too many scenarios

Reply:
Unfortunately `FileNotFoundError` is not defined Python 2. IOError handles the case that the file isn't found and that the file is unreadable for some other reason.

Reply:
I wonder if I should check if the file exists and handle the IOError separately?

Reply:
You want different behaviors for a non-existing public file vs other unreadable file errors(no permission, password-protected, etc) correct?

In that case, you need to differentiate between the two; if FileNotFound is not available in Python2, the best option seems to simply check for existence.

Reply:
I agree

Reply:
I think I have addressed your comments, thanks for the feedback.